app/controllers/concerns/calendarable.rb

```diff
@@ -0,0 +1,94 @@
+module Calendarable
+  extend ActiveSupport::Concern
+
```

Comment:
Good idea - I documented those methods below but it makes sense to consolidate. I'm going to clean up this file and then have you take a look over the comments before we merge.